tests/test_csc.py

```diff
@@ -0,0 +1,401 @@
+                )
+
+    async def assert_dome_elevation(self, expected_elevation, move_expected):
+        """Check the Dome and MTDomeController commanded elevation.
+
+        Parameters
+        ----------
+        expected_elevation : `float`
+            Expected new elevation position (deg);
+            ignored if ``move_expected`` false.
+        move_expected : `bool`
+            Is a move expected?
+
+        Notes
+        -----
+        If ``move_expected`` then read one ``elTarget`` Dome event.
+        """
+        if move_expected:
+            dome_elevation_target = await self.dome_remote.evt_elTarget.next(
+                flush=False, timeout=STD_TIMEOUT
+            )
+            salobj.assertAnglesAlmostEqual(
+                dome_elevation_target.position, expected_elevation
+            )
+        else:
+            with self.assertRaises(asyncio.TimeoutError):
+                await self.dome_remote.evt_elTarget.next(
+                    flush=False, timeout=NODATA_TIMEOUT
+                )
+
+    def assert_telescope_target(self, expected_elevation, expected_azimuth):
+        salobj.assertAnglesAlmostEqual(
+            self.csc.telescope_target.elevation.position, expected_elevation
+        )
+        salobj.assertAnglesAlmostEqual(
+            self.csc.telescope_target.azimuth.position, expected_azimuth
+        )
+
+    def scaled_maz_delta_azimuth(self, elevation):
+        """max_delta_azimuth scaled by cos(elevation).
+
+        Thus the minimum azimuth difference that will trigger a dome move
+        for the simple algorithm.
+        """
+        return self.csc.algorithm.max_delta_azimuth / math.cos(elevation * RAD_PER_DEG)
+
+    async def check_move(
+        self, elevation, azimuth, move_elevation, move_azimuth, wait_dome_done
+    ):
+        """Set telescope target azimuth and elevation.
+
+        Check that the dome moves there in azimuth or elevation,
+        as requested.
+        Then check that the dome does not move for small changes
+        to the telescope target about that point.
+
+        Parameters
+        ----------
+        elevation : `float`
+            Desired elevation for telescope (deg)
+        azimuth : `float`
+            Desired azimuth for telescope and dome (deg)
+        move_elevation : `bool`
+            Move the dome in elevation?
+        move_azimuth : 'bool`
+            Move the dome in azimuth?
+        wait_dome_done : `bool`
+            Wait for the dome move to finish?
+
+        Raises
+        ------
+        ValueError :
+            If the change in dome azimuth <= configured max dome azimuth error
+            (since that will result in no dome motion, which will mess up
+            the test).
+        """
+        print(
+            f"check_move: elevation={elevation}, azimuth={azimuth}; "
+            f"move_elevation={move_elevation}, move_azimuth={move_azimuth}; "
+            f"wait_dome_done={wait_dome_done}"
```

Comment:
Maybe less relevant for unit tests but do you really want to use a print statement here and not a log statement?

Reply:
I prefer print statements in unit tests because I know they will be printed.